Engine/Source/ContentLoader.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 Genten Studios
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <Quartz2/ContentLoader.hpp>
+
+using namespace q2;
+
+Mod::Mod(std::string name)
+    : m_name(std::move(name))
+{
+    std::fstream fileStream;
+    fileStream.open("Modules/" + name + "/dependencies.txt");
+    while (fileStream.peek() != EOF){
+        std::string input;
+        std::getline(fileStream, input);
+        m_dependencies.push_back(input);
+    }
+    fileStream.close();
+};
+Mod::~Mod(){};
+
+bool Mod::exists(){
+    return true; //TODO make this actually check for mod
+}
+
+bool modules::loadModules(std::string save, sol::state& lua)
+{
+    std::fstream fileStream;
+    std::queue<Mod> toLoad; //A queue of mods that need loaded
+
+    fileStream.open("Save/" + save + "/mods.txt");
+    int i = 0;
+    while (fileStream.peek() != EOF){
+        std::string input;
+        std::getline(fileStream, input);
+        Mod mod = Mod(input);
+        if(!mod.exists()){std::cout << "Mod does not exist" + mod.m_name; return false;}; //Should never happen if launcher does all the work but lets check anyways
```

Comment:
Could you move the body of this if statement onto their own lines please?
Also it would probably be better to use the quartz logging utilities
 - It may be the case that these have not been copied over yet, if so we should probably do that before this gets merged.
```suggestion
        if(!mod.exists())
        {
                // Should never happen if launcher does all the work but lets check anyways
                std::cout << "Mod does not exist" + mod.m_name;
                return false;
        }
```